configs/components/fesom/fesom-2.1.yaml

```diff
@@ -38,6 +38,75 @@ choose_version:
     wiso_code: true
     git-repository:
     - https://gitlab.dkrz.de/FESOM/fesom2.git
+  2.1-recom-par_tracers: # MA: for Kai testing ESM-Tools in Levante
+    branch: feature/par_tracers
+    git-repository:
+    - https://gitlab.dkrz.de/FESOM/fesom2.git
+    destination: fesom-2.1-recom-par_tracers
+
+  2.1-recom-par_tracers_mpi: # kh 13.04.22 for testing FESOM-REcoM MPI based tracer loop parallelization with ESM-Tools in Levante
+    branch: awiesm2-recom-parallel
+    git-repository:
+    - https://gitlab.dkrz.de/FESOM/fesom2.git
+    destination: fesom-2.1-recom-par_tracers_mpi
+# kh 08.03.22 todo: merge conflict (if enabled)
+#   include_models:
+#       - recom
+# YY: 29.04.22 to compile wiso code
+    wiso_code: true
+
+# MA: changes specific to PAR-TRACERS go here so they don't destroy other working versions of FESOM-2.1)
+# TODO: remove this block when PAR-TRACERS is generalized!
+par_tracers: "$(( 'par_tracers' in  '${fesom.version}' ))"
+choose_fesom.par_tracers:
+    True:
+        # kh 21.02.22 use uppercase here
+        forcing_data_dir: "${pool_dir}/FORCING/"
+        # kh 21.02.22
+        asforcing: CORE2
+        namelists:
+                - namelist.config
+                - namelist.forcing
+                - namelist.oce
+                - namelist.ice
+                - namelist.io
+        # kh 21.02.22 temporary disabled to get fesom-2.1-recom-par_tracers running on levante
+        #       - namelist.cvmix
```

Comment:
Same as above; we need here some generalised solution, or ideally a consolidated FESOM codebase.

Reply:
For a general solution of the namelist reference problem I see two possibilities:
a) A consolidated FESOM code base (as already suggested by Paul).

b) A support for adding/removing namelist references via an appropriate data structure, replacing the current commenting in or out of the namelist references in the yaml code. If, for example, a base namelist is maintained that initially contains all possible namelist references, the appropriate deletion of unneeded references would be sufficient for a concrete FESOM variant. If necessary, more sophisticated tree-like structures would also be conceivable, so that common namelist references of similar FESOM variants could also be factored out (like in a typical object-oriented class design).

Variant b) probably offers more potential for future similar cases but variant a) seems better maintainable and more stable.

Reply:
Option b already exists. I'll commit a change using the `remove` feature.

Reply:
This is how it looks now and how one can use `remove_` to remove elements of dictionaries and lists:

https://github.com/esm-tools/esm_tools/blob/e0d47a21bbab3c5ed8653c871be2e195bc046db9/configs/components/fesom/fesom-2.1.yaml#L68-L78